apps/ws/src/shared/framework/in-memory-io.adapter.ts

```diff
@@ -0,0 +1,38 @@
+import { IoAdapter } from '@nestjs/platform-socket.io';
+import { ServerOptions } from 'socket.io';
+import Redis from 'ioredis';
+import { createAdapter } from '@socket.io/redis-adapter';
+import { getRedisPrefix } from '@novu/shared';
+import { WebSocketsInMemoryProviderService } from '@novu/application-generic';
+import { ConnectionOptions } from 'tls';
+
+export class InMemoryIoAdapter extends IoAdapter {
+  private webSocketsInMemoryProviderService: WebSocketsInMemoryProviderService;
+  private adapterConstructor: ReturnType<typeof createAdapter>;
+
+  async connectToInMemoryCluster(): Promise<void> {
+    // TODO: Pending to inject in the provider instantiation
+    const keyPrefix = getRedisPrefix() ? `socket.io#${getRedisPrefix()}` : 'socket.io';
+
+    this.webSocketsInMemoryProviderService = new WebSocketsInMemoryProviderService();
+
+    await this.webSocketsInMemoryProviderService.initialize();
+
+    const pubClient = this.webSocketsInMemoryProviderService.getClient();
+    const subClient = pubClient?.duplicate();
+
+    /*
+     *  TODO: Might not be needed to connect as we are checking it is initialized already.
+     * await Promise.all([pubClient?.connect(), subClient?.connect()]);
```

Comment:
This code was existing before my changes. If I leave it, now because the InMemoryProviderService downstream is initialised if you try to connect again it errors.